src/amqp.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#ifndef  _GNU_SOURCE
+#define  _GNU_SOURCE
+#endif
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <ell/ell.h>
+#include <hal/linux_log.h>
+#include <amqp.h>
+#include <amqp_tcp_socket.h>
+
+#include "settings.h"
+#include "amqp.h"
+
+#define AMQP_CONNECTION_TIMEOUT_US 10000
+
+static amqp_connection_state_t conn;
+static struct amqp_connection_info cinfo;
+
+static const char *amqp_server_exception_string(amqp_rpc_reply_t reply)
+{
+	amqp_connection_close_t *m = reply.reply.decoded;
+	static char r[512];
+
+	switch (reply.reply.id) {
+	case AMQP_CONNECTION_CLOSE_METHOD:
+		snprintf(r, sizeof(r),
+			 "server connection error %uh, message: %.*s\n",
+			 m->reply_code, (int)m->reply_text.len,
+			 (char *)m->reply_text.bytes);
+		break;
+	case AMQP_CHANNEL_CLOSE_METHOD:
+		snprintf(r, sizeof(r),
+			 "server channel error %uh, message: %.*s\n",
+			 m->reply_code, (int)m->reply_text.len,
+			 (char *)m->reply_text.bytes);
+		break;
+	default:
+		snprintf(r, sizeof(r),
+			 "unknown server error, method id 0x%08X\n",
+			 reply.reply.id);
+		break;
+	}
+
+	return l_strdup(r);
+}
+
+static const char *amqp_rpc_reply_string(amqp_rpc_reply_t reply)
+{
+	switch (reply.reply_type) {
+	case AMQP_RESPONSE_NONE:
+		return "missing RPC reply type!";
+	case AMQP_RESPONSE_LIBRARY_EXCEPTION:
+		return amqp_error_string2(reply.library_error);
+	case AMQP_RESPONSE_SERVER_EXCEPTION:
+		return amqp_server_exception_string(reply);
+	case AMQP_RESPONSE_NORMAL:
+	default:
+		return "";
+	}
+}
+
+int amqp_start(struct settings *settings)
+{
+	amqp_socket_t *socket;
+	amqp_rpc_reply_t r;
+	struct timeval timeout = { .tv_usec=AMQP_CONNECTION_TIMEOUT_US };
+	int status;
+
+	status = amqp_parse_url((char*) settings->rabbitmq_url, &cinfo);
+	if (status) {
+		hal_log_error("amqp_parse_url: %s", amqp_error_string2(status));
+		return -1;
+	}
+
```

Comment:
Why not logging that you're trying to connect?

Reply:
I don't think that is good to log it. If there a connection error will be printed and if not a message successful. Maybe this log can fit with a `hal_log_debug`.
What do you think?

Reply:
I agree with you. I thing hal_log_debug would fit better